include/oneapi/mkl/dft/backward.hpp

```diff
@@ -26,51 +26,105 @@
 #include <CL/sycl.hpp>
 #endif
 
+#include "detail/types_impl.hpp"
+
 namespace oneapi::mkl::dft {
 //Buffer version
 
 //In-place transform
 template <typename descriptor_type, typename data_type>
-void compute_backward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout);
+void compute_backward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout) {
+    using fwd_type = typename detail::descriptor_info<descriptor_type>::forward_type;
+    auto type_corrected_inout = inout.template reinterpret<fwd_type, 1>(
```

Comment:
In quite a few places, buffers are reinterpreted to a "correct" type without checking the validity of the input types. I think more type checking is needed here. Eg.
* A user might be lead to expect that eg. passing single-precision data to a double precision transform is a supported operation.
* A user might want errors on passing in eg. integers.

Reply:
added in https://github.com/oneapi-src/oneMKL/pull/342/commits/a8a004c57696f512cc9f9e54aab3aaea0bedd536